packages/modules/product/src/repositories/product-category.ts

```diff
@@ -158,37 +159,34 @@ export class ProductCategoryRepository extends DALUtils.MikroOrmBaseTreeReposito
 
     mpaths.push({ mpath: Array.from(parentMpaths) })
 
-    const whereOptions = {
-      ...findOptions.where,
-      $or: mpaths,
-    }
-
-    if ("parent_category_id" in whereOptions) {
-      delete whereOptions.parent_category_id
-    }
+    const where = { ...findOptions.where, $or: mpaths }
+    const options = {
+      ...findOptions.options,
+      limit: undefined,
+      offset: 0,
+    } as MikroOptions<ProductCategory>
 
-    if ("id" in whereOptions) {
-      delete whereOptions.id
-    }
+    delete where.id
+    delete where.mpath
+    delete where.parent_category_id
 
-    let allCategories = await manager.find(
-      ProductCategory,
-      whereOptions as MikroFilterQuery<ProductCategory>,
-      findOptions.options as MikroOptions<ProductCategory>
+    const categoriesInTree = JSON.parse(
```

Comment:
I think instead of doing a clone, we can do `Array.from` on the resulting `Collection`

Reply:
I used `this.serialize` instead. Array.from was throwing some other random errors.